### PR TITLE
need to update balance when op deleted

### PR DESCRIPTION
**Pull request by geopin90**

### CODE REVIEW

File: store/src/elements.rs [hunk 1]
```diff
@@ -160,9 +160,9 @@ pub fn receive_data(
   let mut new_data = data.clone();
   let mut new_before = before.clone();
 
-  if data["status"].string() == "deleted".to_string() {
-    return Ok(old_data);
-  }
+  // if data["status"].string() == "deleted".to_string() {
```
Comment (shabanovd) [inline store/src/elements.rs:163]: delete it

File: store/src/elements.rs [hunk 2]
```diff
@@ -334,24 +334,31 @@ fn json_to_ops(
...
+        }
       };
-      let cost = data["cost"]["number"].number_or_none();
+      let cost =
+        if data["status"] == "deleted" { None } else { data["cost"]["number"].number_or_none() };
 
       if qty.is_none() && cost.is_none() {
-        return Ok(ops);
+        // return Ok(ops);
+        InternalOperation::Receive(0.into(), 0.into())
```
Comment (shabanovd) [inline store/src/elements.rs:350]: why you return something when there is no qty & cost?

File: store/src/elements.rs [hunk 3]
```diff
@@ -334,24 +334,31 @@ fn json_to_ops(
...
+        None
+      } else {
+        match ctx_str[..] {
+          ["production", "produce"] => data["qty"].number_or_none(),
+          _ => data["qty"]["number"].number_or_none(),
+        }
       };
-      let cost = data["cost"]["number"].number_or_none();
+      let cost =
+        if data["status"] == "deleted" { None } else { data["cost"]["number"].number_or_none() };
```
Comment (shabanovd) [inline store/src/elements.rs:346]: why you check status here?

File: store/src/elements.rs [hunk 4]
```diff
@@ -334,24 +334,31 @@ fn json_to_ops(
...
+        // return Ok(ops);
+        InternalOperation::Receive(0.into(), 0.into())
       } else {
         InternalOperation::Receive(qty.unwrap_or_default(), cost.unwrap_or_default().into())
       }
     },
     OpType::Transfer | OpType::Dispatch => {
-      let qty = data["qty"]["number"].number_or_none();
-      let cost = data["cost"]["number"].number_or_none();
+      let qty = if data["status"] == "deleted" { None } else { data["qty"]["number"].number_or_none() };
```
Comment (shabanovd) [inline store/src/elements.rs:356]: why you check status here?

File: store/src/elements.rs [hunk 5]
```diff
@@ -334,24 +334,31 @@ fn json_to_ops(
...
+      let cost = if data["status"] == "deleted" { None } else { data["cost"]["number"].number_or_none() };
 
       if qty.is_none() && cost.is_none() {
-        return Ok(ops);
```
Comment (shabanovd) [inline store/src/elements.rs:354]: this must not be changed

File: store/src/elements.rs (hunk 2 above)
Comment (geopin90) [reply store/src/elements.rs:350]: because I want this operation to be deleted later in mutate_op()

**Review by shabanovd — APPROVED** [review]

**Pull request merged**